src/lib/transaction_logic/mina_transaction_logic.ml

```diff
@@ -952,68 +954,27 @@ module Make (L : Ledger_intf.S) :
           in
           let%bind () =
             Result.ok_if_true
-              (Account.has_permission_to_receive receiver_account)
+              (Account.has_permission_to_send fee_payer_account)
               ~error:Transaction_status.Failure.Update_not_permitted_balance
           in
-          let%bind source_location, source_account =
-            let ret =
-              if Account_id.equal source receiver then
-                (*just check if the timing needs updating*)
-                let%bind location, account =
-                  match receiver_location with
-                  | `Existing _ ->
-                      return (receiver_location, receiver_account)
-                  | `New ->
-                      Result.fail Transaction_status.Failure.Source_not_present
-                in
-                let%map timing =
-                  validate_timing ~txn_amount:amount
```

Comment:
can we keep this block but use `fee_payer` instead of `source`? The receiver timing here is updated and handled accordingly in the snark as well

Reply:
The fee payer's timing is already checked here https://github.com/MinaProtocol/mina/pull/13048/files#diff-cda3a1675db3ca72dfa8ebd6a41737fc11fbd750f764034958ca2acbbf8eb900R937. I didn't think it made sense to keep both timing checks since they are always the same account. But now that you point it out, I'm realizing that it's a mistake since the txn_amount is different across the timing calls. Would it suffice to just update the fee payer timing check to be `~txn_amount:amount` instead of `~txn_amount:Amount.zero`?

Reply:
Oh I misread the code here- source is equal to the receiver in this block. So, we don't have to update receiver's timing if it is different from source. 
>The fee payer's timing is already checked here https://github.com/MinaProtocol/mina/pull/13048/files#diff-cda3a1675db3ca72dfa8ebd6a41737fc11fbd750f764034958ca2acbbf8eb900R937. 

This case is for delegate, no? The amount is zero because fee payer timing is already updated in `pay_fee'` function. So validating it again should be a no-op.

For the payment case, I think checking source timing with `~txn_amount:amount` makes sense because in `pay_fee'` timing is validated with `~txn_amount:fee`. 
>Would it suffice to just update the fee payer timing check to be ~txn_amount:amount instead of ~txn_amount:Amount.zero?

For the fee payer, we should validate timing use the fee (not zero or payment amount) and amount separately because the fee payer may not have the funds to transfer payment amount